core/src/main/java/org/apache/spark/SparkFirehoseListener.java

```diff
@@ -140,6 +140,16 @@ public void onBlockUpdated(SparkListenerBlockUpdated blockUpdated) {
   }
 
   @Override
+  public void onSpeculativeTaskSubmitted(SparkListenerSpeculativeTaskSubmitted speculativeTask) {
+    onEvent(speculativeTask);
+  }
+
+  @Override
+  public void onExtraExecutorNeeded() {
+    onEvent(null);
```

Comment:
`SparkFirehoseListener` is a public API and users may assume `onEvent` will never accept null, how about we just do nothing here?